src/inmanta/data/__init__.py

```diff
@@ -2186,6 +2189,27 @@ def to_dict(self) -> JsonType:
 
         return result
 
+    @classmethod
+    async def execute_in_retryable_transaction(
+        cls, fnc: Callable[[Connection], Awaitable[TransactionResult]], tx_isolation_level: Optional[str] = None
+    ) -> TransactionResult:
+        """
+        Execute the queries in fnc using the transaction isolation level `tx_isolation_level` and return the
+        result returned by fnc. This method performs retries when the transaction is aborted due to a
+        serialization error.
+        """
+        async with cls.get_connection() as postgresql_client:
+            max_retries = 3
+            while True:
+                try:
+                    async with postgresql_client.transaction(isolation=tx_isolation_level):
+                        return await fnc(postgresql_client)
+                except SerializationError:
+                    if max_retries >= 0:
+                        max_retries -= 1
```

Comment:
Shouldn't we include a delay? Markus even recommended an increasing delay but I think we can safely simplify that for testing purposes.

Reply:
Indeed. I didn't include it because it's a test suite here. But since I moved this logic to `src/inmanta/data/__init__.py`, it makes sense to add the backoff logic as well.